pinecone/data/features/bulk_import.py

```diff
@@ -0,0 +1,193 @@
+from enum import Enum
+from typing import Optional, Literal, Iterator, List, Type, cast
+
+from pinecone.config.config import ConfigBuilder
+from pinecone.core_ea.openapi.db_data import ApiClient
+from pinecone.core_ea.openapi.db_data.api.bulk_operations_api import BulkOperationsApi
+from pinecone.core_ea.openapi.shared import API_VERSION
+
+from pinecone.utils import parse_non_empty_args, install_json_repr_override, setup_openapi_client
+
+from pinecone.core_ea.openapi.db_data.models import (
+    StartImportRequest,
+    StartImportResponse,
+    ImportListResponse,
+    ImportModel,
+    ImportErrorMode as ImportErrorModeClass,
+)
+
+for m in [StartImportResponse, ImportListResponse, ImportModel]:
+    install_json_repr_override(m)
+
+ImportErrorMode: Type[Enum] = cast(
+    Type[Enum], Enum("ImportErrorMode", ImportErrorModeClass.allowed_values[("on_error",)])
+)
+
+
+class ImportFeatureMixin:
+    def __init__(self, **kwargs):
+        config = ConfigBuilder.build(
+            **kwargs,
+        )
+        openapi_config = ConfigBuilder.build_openapi_config(config, kwargs.get("openapi_config", None))
+
+        if kwargs.get("__import_operations_api", None):
+            self.__import_operations_api = kwargs.get("__import_operations_api")
+        else:
+            self.__import_operations_api = setup_openapi_client(
+                api_client_klass=ApiClient,
+                api_klass=BulkOperationsApi,
+                config=config,
+                openapi_config=openapi_config,
+                pool_threads=kwargs.get("pool_threads", 1),
+                api_version=API_VERSION,
+            )
+
+    def start_import(
+        self,
+        uri: str,
+        integration_id: Optional[str] = None,
+        error_mode: Optional[Literal["CONTINUE", "ABORT"]] = "CONTINUE",
+    ) -> StartImportResponse:
+        """Import data from a storage provider into an index. The uri must start with the scheme of a supported
+        storage provider. For buckets that are not publicly readable, you will also need to separately configure
+        a storage integration and pass the integration name.
+
+        Examples:
+            >>> from pinecone import Pinecone
+            >>> index = Pinecone().Index('my-index')
+            >>> index.start_import(uri="s3://bucket-name/path/to/data.parquet")
+            { id: "1" }
+
+        Args:
+            uri (str): The URI of the data to import. The URI must start with the scheme of a supported storage provider.
+            integration (Optional[str], optional): If your bucket requires authentication to access, you need to pass the name of your storage integration using this property. Defaults to None.
+            error_mode: Defaults to "CONTINUE". If set to "CONTINUE", the import operation will continue even if some
```

Comment:
idk if you want to do this, but in the TS client I lowercase whatever they input just to avoid imports failing b/c users put "Continue" vs "CONTINUE"